### PR TITLE
Clean some unwraps

### PR DESCRIPTION
**Pull request by Hoverbear**
Removes some unnecessary unwraps and clarifies why some others are safe.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by eeyun** [comment]
Awesome! Looks good to me but I'd like to get another set of eyes on this. I'm 100% for removing unnecessary unwraps.

File: components/hab/src/gossip.rs [hunk 1]
```diff
@@ -259,12 +259,7 @@ pub mod hab_gossip {
         pub fn reset_heat_for(&mut self, rumor_id: &RumorId) {
             for (_member_id, mut rumor_heat) in self.heat.iter_mut() {
                 debug!("Reset heat for {:?}", rumor_id);
-                if rumor_heat.contains_key(rumor_id) {
-                    let mut count = rumor_heat.get_mut(rumor_id).unwrap();
-                    *count = 0;
-                } else {
-                    rumor_heat.insert(rumor_id.clone(), 0);
-                }
+                *rumor_heat.entry(*rumor_id).or_insert(0) = 0;
```
Comment (Hoverbear) [inline components/hab/src/gossip.rs]: This particular change I'm not **entirely** sure has the same behavior. It should, but it's the first time I've used this scheme myself.

**Comment by reset** [comment]
@thesentinels approve

gif-keyboard-11714087522844893858

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/172049726).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/172049726).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**